### PR TITLE
fix(list): reorganize props

### PR DESCRIPTION
**Pull request by jsomsanith**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
The list props become too heavy, with a lot of things in list props root.

**What is the new behavior?**
Now list props have those props
*Data*:
* id
* columns
* items

*Part configuration*
* titleProps : title configuration
* itemProps : single item configuration (except onToggleAll that should be moved to toolbar)

**Does this PR introduce a breaking change?**

- [x] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: stories/List.js [hunk 1]
```diff
@@ -81,18 +73,28 @@ const props = {
...
+				let found = false;
+				for (let i = 0; i < selected.length; i += 1) {
+					if (selected[i].id === item.id) {
+						found = true;
+						break;
+					}
 				}
-			}
-			return found;
+				return found;
```
Comment (mtikhinskaya) [inline stories/List.js]: How about `return selected.findIndex(el => el.id === item.id) > -1;` ?

File: package.json [hunk 2]
```diff
@@ -99,7 +99,7 @@
     "react": "^15.4.0",
     "react-addons-css-transition-group": "^15.4.1",
     "react-bootstrap": "0",
-    "talend-icons": "^0.5.5",
+    "talend-icons": "0.8.0",
```
Comment (jsomsanith) [inline package.json]: Fixed it, thanks

File: stories/List.js (hunk 1 above)
Comment (jsomsanith) [inline stories/List.js]: Changed by Array.find()

**Comment by jsomsanith** [comment]
@jmfrancois not a chance :)
But I think now the props are a little clearer, this component is the more complex we have, and it is hard to use, this will help us I think.
The props will change again when we export the `select all` checkbox to the toolbar

**Pull request merged**